x/mongo/driver/topology/topology.go

```diff
@@ -234,7 +236,22 @@ func (t *Topology) Connect() error {
 
 	t.serversLock.Unlock()
 	if t.pollingRequired {
-		go t.pollSRVRecords()
+		uri, err := url.Parse(t.cfg.URI)
```

Comment:
I'm concerned we don't have enough test coverage of MongoDB connection string formats in the SRV polling tests to assert that it's safe to use `url.Parse`.

Is every `mongodb+srv://` connection string parseable with `url.Parse`?

Reply:
The [package doc](https://pkg.go.dev/net/url#URL) says `url.Parse` parses `[scheme:][//[userinfo@]host][/]path[?query][#fragment]`.
It has been proven that `mongodb+srv` is parseable as a legal scheme, and covered in the test cases.
Mongo host in the connection string may be different from ordinary hostnames, e.g. a comma-separated hosts list, which I can add as a test case, but should be illegal for SRV still.
Also, we don't need to parse the `path` string in this case so it doesn't impact us even though we may have special cases in the query parameters.

Reply:
Sounds good. I also tested out using `url.Parse` to parse a bunch of connection string formats and they all worked correctly (see [here](https://go.dev/play/p/pesmu2YBNhH)).